src/parser/GenericParser.ts

```diff
@@ -5,14 +5,32 @@ import { Configuration } from "./Configuration";
 import { MetricCalculator } from "./MetricCalculator";
 import { CouplingCalculator } from "./CouplingCalculator";
 import { CouplingResult, MetricResult } from "./metrics/Metric";
+import { debuglog } from "node:util";
 
+const dlog = debuglog("metric-gardener");
```

Comment:
This is going to return a wrapper to log things. If the logging time itself should also be reduced, it's possible to use the callback. That way the additional overhead should be lower.

https://nodejs.org/api/util.html#utildebuglogsection-callback

Reply:
Ok, thanks for the hint. I have changed that.